cdk/lib/payment-api.ts

```diff
@@ -1,22 +1,409 @@
+import {
+  GuAllowPolicy,
+  GuGetS3ObjectsPolicy,
+  GuPutCloudwatchMetricsPolicy,
+} from "@guardian/cdk/lib/constructs/iam";
+import type { GuAsgCapacity } from "@guardian/cdk/lib/types";
+
+interface PaymentApiProps extends GuStackProps {
+  domainName: string;
+  scaling: GuAsgCapacity;
+}
 
 export class PaymentApi extends GuStack {
-  constructor(scope: App, id: string, props: GuStackProps) {
+  constructor(scope: App, id: string, props: PaymentApiProps) {
     super(scope, id, props);
+
+    const emailSqsCodeArn = new GuStringParameter(
+      this,
+      "EmailSqsQueueCodeArn",
+      {
+        description:
+          "For the PROD stack you still need to supply this, because PROD instances need access to both PROD and CODE email queues.",
+      }
+    );
+
+    const emailSqsProdArn = new GuStringParameter(
+      this,
+      "EmailSqsQueueProdArn",
+      {
+        description:
+          "For the CODE stack you can leave this empty since it won't be used.  For the PROD stack you need to set it.",
+      }
+    );
+
+    const ophanRole = new GuStringParameter(this, "OphanRole", {
+      description: "ARN of the Ophan cross-account role",
+    });
+
+    const kinesisStreamArn = new GuStringParameter(this, "KinesisStreamArn", {
+      description: "ARN of the kinesis stream to write events to",
+    });
+
+    const contributionsStoreSqsQueueCodeArn = new GuStringParameter(
+      this,
+      "ContributionsStoreSqsQueueCodeArn",
+      {
+        description:
+          "For the PROD stack you still need to supply this, because PROD instances need access to both PROD and CODE contribution store queues.",
+      }
+    );
+
+    const contributionsStoreSqsQueueProdArn = new GuStringParameter(
+      this,
+      "ContributionsStoreSqsQueueProdArn",
+      {
+        description:
+          "For the CODE stack you can leave this empty since it won't be used.  For the PROD stack you need to set it.",
+      }
+    );
+
+    const sqsKmsArn = new GuStringParameter(this, "SqsKmsArn", {
+      description: "ARN of the KMS key for encrypting SQS data",
+    });
+
     const yamlTemplateFilePath = join(
       __dirname,
       "../..",
       "support-payment-api/src/main/resources/cloud-formation.yaml"
     );
+
     new CfnInclude(this, "YamlTemplate", {
       templateFile: yamlTemplateFilePath,
       parameters: {
         Stage: props.stage,
+        EmailSqsQueueCodeArn: emailSqsCodeArn,
+        EmailSqsQueueProdArn: emailSqsProdArn,
+        OphanRole: ophanRole,
+        KinesisStreamArn: kinesisStreamArn,
+        ContributionsStoreSqsQueueCodeArn: contributionsStoreSqsQueueCodeArn,
+        ContributionsStoreSqsQueueProdArn: contributionsStoreSqsQueueProdArn,
+        SqsKmsArn: sqsKmsArn,
+      },
+    });
+
+    // TODO: Should these remain as cloudformation parameters?
+    const projectName = "payment-api";
+    const projectVersion = "0.1";
```

Comment:
Currently these are cloudformation parameters, but I'm not sure how much value that adds, so we've just hardcoded them for now.